internal/gengapic/genrest.go

```diff
@@ -67,25 +76,78 @@ func (g *generator) genRESTMethods(serv *descriptor.ServiceDescriptorProto, serv
 	return nil
 }
 
+func (g *generator) restClientOptions(serv *descriptor.ServiceDescriptorProto, servName string) error {
+	p := g.printf
+
+	var host string
+	if eHost, err := proto.GetExtension(serv.Options, annotations.E_DefaultHost); err == nil {
+		host = *eHost.(*string)
+	} else {
+		fqn := g.descInfo.ParentFile[serv].GetPackage() + "." + serv.GetName()
+		return fmt.Errorf("service %q is missing option google.api.default_host", fqn)
+	}
+
+	p("func default%sRESTClientOptions() []option.ClientOption {", servName)
+	p("  return []option.ClientOption{")
+	p("    internaloption.WithDefaultEndpoint(%q),", host)
```

Comment:
Do library users have the ability to manually set the protocol to be HTTP rather than HTTPS at client instantiation time? If not, until we do have such an ability, would this be the right place to automatically set it to be HTTP if the endpoint starts with `localhost`? (This would be a stop-gap measure until we can have users do it at client creation time; I can point you to the internal doc on the topic. Or see the Python PR....)

This would be needed to hook up the Showcase integration tests; we can make those two changes at the same time.

Reply:
At least in apiary land the default endpoints contain the scheme as well. So if a user wants non-https they just specify when they override it.

Example : 
- `internaloption.WithDefaultEndpoint(https://secretmanager.googleapis.com/)`
- ``internaloption.WithDefaultMTLSEndpoint(https://secretmanager.mtls.googleapis.com/)

Reply:
So this is something we'll be able to do immediately for REST clients at instantiation time, once this PR is merged? Perfect.